src/fairchem/core/scripts/diatomic_tests.py

```diff
@@ -0,0 +1,216 @@
+
+mol_dict = {
+    ("C", "C"): {
+        "atoms_info": {"spin": 1, "charge": 0},
+        "ref_energy": -2059.708257873304,
+    },
+    ("H", "H"): {
+        "atoms_info": {"spin": 1, "charge": 0},
+        "ref_energy": -26.89116811474675,
+    },
+    ("O", "O"): {
+        "atoms_info": {"spin": 3, "charge": 0},
+        "ref_energy": -4085.9570373686342,
+    },
+    ("N", "N"): {
+        "atoms_info": {"spin": 1, "charge": 0},
+        "ref_energy": -2971.0840536105143,
+    },
+    ("F", "F"): {
+        "atoms_info": {"spin": 1, "charge": 0},
+        "ref_energy": -5428.480814580473,
+    },
+    ("S", "S"): {
+        "atoms_info": {"spin": 3, "charge": 0},
+        "ref_energy": -21665.24472704541,
+    },
+}
+
+mat_dict = {
+    ("C", "C"): {"atoms_info": {}, "ref_energy": -2.55503062},
+    ("O", "O"): {"atoms_info": {}, "ref_energy": -3.09594272},
+    ("Pt", "Pt"): {"atoms_info": {}, "ref_energy": -1.00580368},
+    ("Si", "Si"): {"atoms_info": {}, "ref_energy": -1.65238266},
+    ("Cu", "Cu"): {"atoms_info": {}, "ref_energy": -0.46898334},
+    ("Mg", "Mg"): {"atoms_info": {}, "ref_energy": -0.01886358},
+    ("Fe", "O"): {"atoms_info": {}, "ref_energy": -4.85459983},
+    ("Ca", "O"): {"atoms_info": {}, "ref_energy": -1.57365455},
+}
+
+
+def get_model(
+    model_name,
+    released,
+):
+    if not released:
+        uma_pred = pretrained_mlip.load_predict_unit(
+            model_name,
+            device="cuda",
+        )
+    else:
+        uma_pred = pretrained_mlip.get_predict_unit(model_name, device="cuda")
+    return uma_pred
+
+
+def compute_diatomic_curve(predictor, task, atom_1, atom_2, atoms_dict):
+    # Get the atomic number
+    atomic_number_1 = atomic_numbers[atom_1]
+    atomic_number_2 = atomic_numbers[atom_2]
+
+    rmin = 0.5 * (covalent_radii[atomic_number_1] + covalent_radii[atomic_number_2])
+    rmax = 6.0
+
+    distances = np.linspace(rmin, rmax, 100)  # Range of distances in Angstroms
+    energies = []
+
+    # Calculate energy for each distance
+    atoms_list = []
+    for d in distances:
+        # Create the diatomic molecule
+        dimer = Atoms([atom_1, atom_2], positions=[[0, 0, 0], [0, 0, d]], pbc=False)
+        if atoms_dict["atoms_info"] is not None:
+            dimer.info.update(atoms_dict["atoms_info"])
+
+        atoms_list.append(dimer)
+
+    if task == "omol":
+        atomic_data_list = [
+            AtomicData.from_ase(atoms, task_name="omol", r_data_keys=["spin", "charge"])
+            for atoms in atoms_list
+        ]
+    else:
+        atomic_data_list = [
+            AtomicData.from_ase(
+                atoms,
+                task_name=task,
+            )
+            for atoms in atoms_list
+        ]
+    batch = atomicdata_list_to_batch(atomic_data_list)
+
+    energies = predictor.predict(batch)["energy"]
+
+    # reference energy is the sum of the individual atom energies
+    relative_energies = energies.detach().cpu().numpy() - atoms_dict["ref_energy"]
```

Comment:
Are these ref energies in atoms_dict taken from the respective element_refs for omol and omat? Might be easier to use `predictor.predict(batch, undo_element_references=False)`

Reply:
We discussed offline. The way it is currently written, model predicts DFT total energy then the sum of isolated atomic energies are subtracted off for the reference (i.e. at >6A the model should return the sum of two isolated atom energies). We decided to leave it like this and add to TODO around the hard coded references.